azuread/resource_group_member_test.go

```diff
@@ -0,0 +1,224 @@
+			{
+				Config: testAccAzureADGroupMember_ServicePrincipal(id),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "group_object_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "member_object_id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureADGroupMemberDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azuread_group_member" {
+			continue
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).groupsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		groupID := rs.Primary.Attributes["group_object_id"]
+		memberID := rs.Primary.Attributes["member_object_id"]
+
+		members, err := client.GetGroupMembersComplete(ctx, groupID)
+		if err != nil {
+			if ar.ResponseWasNotFound(members.Response().Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		var memberObjectID string
+		for members.NotDone() {
+			// possible members are users, groups or service principals
+			// we try to 'cast' each result as the corresponding type and diff
+			// if we found the object we're looking for
+			user, _ := members.Value().AsUser()
+			if user != nil {
+				if *user.ObjectID == memberID {
+					memberObjectID = *user.ObjectID
+					// we successfully found the directory object we're looking for, we can stop looping
+					// through the results
+					break
+				}
+			}
+
+			group, _ := members.Value().AsADGroup()
+			if group != nil {
+				if *group.ObjectID == memberID {
+					memberObjectID = *group.ObjectID
+					// we successfully found the directory object we're looking for, we can stop looping
+					// through the results
+					break
+				}
+			}
+
+			servicePrincipal, _ := members.Value().AsServicePrincipal()
+			if servicePrincipal != nil {
+				if *servicePrincipal.ObjectID == memberID {
+					memberObjectID = *servicePrincipal.ObjectID
+					// we successfully found the directory object we're looking for, we can stop looping
+					// through the results
+					break
+				}
+			}
+
+			err = members.NextWithContext(ctx)
+			if err != nil {
+				return fmt.Errorf("Error listing Azure AD Group Members: %s", err)
+			}
+		}
+
+		if memberObjectID != "" {
+			return fmt.Errorf("Azure AD group member still exists:\n%#v", memberObjectID)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureADGroupMember_User(id string, password string) string {
+	return fmt.Sprintf(`
+
+data "azuread_domains" "tenant_domain" {
+	only_initial = true
+}
+
+resource "azuread_user" "test" {
+	user_principal_name   = "acctestA%[1]s@${data.azuread_domains.tenant_domain.domains.0.domain_name}"
+	display_name          = "acctestA%[1]s"
+	password              = "%[2]s"
+}
+	
+resource "azuread_group" "test" {
+	name = "acctest%[1]s"
+}
+
+resource "azuread_group_member" "test" {
+	group_object_id 	= "${azuread_group.test.id}"
```

Comment:
I have added an object_id property to all resources, so could we use that instead to be explicit about what we are using here?
```suggestion
	group_object_id 	= "${azuread_group.test.object_id}"
```